### DISCUSSION TITLE
Why is the result of slint viewer different from vscode, and the rendering result of rust the same as that of slint viewer?

### DISCUSSION THREAD

**Discussion by bleedingfight** [slint-ui/slint#5179] [Q&A] [answered]
- slint:1.5.1
- vscode:1.89.0
- rust：rustc 1.77.2
- os:Linux  6.6.26-1-MANJARO #1 SMP PREEMPT_DYNAMIC Wed Apr 10 20:11:08 UTC 2024 x86_64 GNU/Linux
Vscode:
![image](https://github.com/slint-ui/slint/assets/11495161/9eb05d4a-bd5d-411b-85cc-05187ed4b54b)

slint-viewer(same with rust):
![image](https://github.com/slint-ui/slint/assets/11495161/c8a77cbc-254d-44ec-92b0-ee70402c4d01)

**Comment by ogoffart** [comment]
I believe the difference is coming from the Style in use.
The viewer might use the Qt style while the vs code extension doesn't.

**Reply by bleedingfight** [reply]
@ogoffart 
How to achieve the same result as vscode? This is important to me because the output of the slint-viewer is more targeted and can be seen during debugging.

**Reply by hunger** [reply]
Fastest way is to adjust the Style dropdown in the Live-Preview.

**Reply by ogoffart** [reply, marked as answer]
vscode 's lsp isn't compiled with the Qt style, so that won't work.

But you can disable Qt backend with feature flag, or use other way to set the style, see https://releases.slint.dev/1.5.1/docs/slint/src/advanced/style#selecting-a-widget-style

**Reply by bleedingfight** [reply]
The rendering style displayed by vscode is native, and the rendering effect is similar after setting the style in the slim viewer.
`slint-viewer --style native ui/hyper.slint`
